include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
At some point we'll need to classify some of these algorithms so they can't be used inappropriately. For example an attempt to use a KDF to verify signatures or a non-PBE KDF where a PBE is required. I'd regarded a PBE as a subset of a KDF which always accepts a password parameter.

At some point I'd like to add ASN.1 support to PBEs so they can be handled entirely through EVP_PKEY instead of the various different APIs we currently have. That would tidy up things like PKCS#8 handling quite a bit.